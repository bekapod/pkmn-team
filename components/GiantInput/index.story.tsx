// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from "@storybook/react";
import React from "react";
import GiantInput from ".";

storiesOf("GiantInput", module).add("default", () => (
  <GiantInput type="text" />
));
