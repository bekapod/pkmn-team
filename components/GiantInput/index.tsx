import styled from "styled-components/macro";
import { FlattenSimpleInterpolation } from "styled-components";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";
import TextInput from "../TextInput";
import { media } from "../../helpers/media";

interface Props {
  fullWidth?: boolean;
}

const GiantInput = styled(TextInput)`
  width: 100%;
  padding: 0 ${variables.spacing.md}px;
  height: ${variables.spacing.xl}px;
  font-size: ${variables.fontSizes.lg}px;
  line-height: ${lineHeight("lg")};
  text-align: center;

  ${({ fullWidth }: Props): string | FlattenSimpleInterpolation =>
    fullWidth ? "" : media.medium`max-width: 600px;`}
`;

export default GiantInput;
