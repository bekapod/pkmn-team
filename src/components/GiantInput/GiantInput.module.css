.base {
  width: 100%;
  padding: 0 var(--spacing-md);
  height: var(--spacing-8);
  font-size: var(--font-size-lg);
  text-align: center;

  @media (--medium-viewport) {
    max-width: 600px;
  }
}

.is-full-width {
  @media (--medium-viewport) {
    max-width: none;
  }
}
