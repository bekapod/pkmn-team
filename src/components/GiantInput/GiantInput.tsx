import { forwardRef, FunctionComponent } from 'react';
import cx from 'classnames';
import { TextInput, TextInputProps } from '../TextInput';
import styles from './GiantInput.module.css';

export type GiantInputProps = TextInputProps & {
  fullWidth?: boolean;
};

export const GiantInput: FunctionComponent<GiantInputProps> = forwardRef(
  ({ fullWidth, className, ...props }, ref) => (
    <TextInput
      ref={ref}
      className={cx(styles.base, className, {
        [styles['is-full-width']]: fullWidth
      })}
      {...props}
    />
  )
);
