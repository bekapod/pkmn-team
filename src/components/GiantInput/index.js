// @flow
import styled from "styled-components/macro";
import TextInput from "../TextInput";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";

const GiantInput = styled(TextInput)`
  width: 600px;
  max-width: 100%;
  padding: 0 ${variables.spacing.md}px;
  height: ${variables.spacing.xl}px;
  font-size: ${variables.fontSizes.lg}px;
  line-height: ${lineHeight("lg")};
  text-align: center;
`;

export default GiantInput;
