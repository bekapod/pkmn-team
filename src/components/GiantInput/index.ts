import styled from "styled-components/macro";
import * as variables from "../../helpers/variables";
import { lineHeight } from "../../helpers/verticalRhythm";
import TextInput from "../TextInput";

const GiantInput = styled(TextInput)`
  width: 100%;
  max-width: 600px;
  padding: 0 ${variables.spacing.md}px;
  height: ${variables.spacing.xl}px;
  font-size: ${variables.fontSizes.lg}px;
  line-height: ${lineHeight("lg")};
  text-align: center;
`;

export default GiantInput;
