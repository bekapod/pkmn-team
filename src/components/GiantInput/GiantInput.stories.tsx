import { Meta, Story } from '@storybook/react/types-6-0';
import { GiantInput, GiantInputProps } from './GiantInput';

export default {
  title: 'Components/GiantInput',
  component: GiantInput
} as Meta<GiantInputProps>;

export const Standard: Story<GiantInputProps> = args => (
  <GiantInput {...args} />
);
export const WithPlaceholder: Story<GiantInputProps> = args => (
  <GiantInput placeholder="Some placeholder text" {...args} />
);
