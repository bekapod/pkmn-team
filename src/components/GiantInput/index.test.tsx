import React from "react";
// tslint:disable-next-line:no-implicit-dependencies
import renderer from "react-test-renderer";
import GiantInput from ".";

describe("<GiantInput />", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<GiantInput type="text" />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
