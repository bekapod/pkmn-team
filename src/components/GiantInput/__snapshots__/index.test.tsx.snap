// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiantInput /> renders without crashing 1`] = `
<input
  className="TextInput-sc-149814h-0 GiantInput-sc-1mhmfj5-0 CEtIK"
  type="text"
/>
`;
